src/Microsoft.ML.Data/Scorers/ScoreMapperSchema.cs

```diff
@@ -4,327 +4,106 @@
-                TextType.Instance.GetPair(MetadataUtils.Kinds.ScoreColumnKind),
-                TextType.Instance.GetPair(MetadataUtils.Kinds.ScoreValueKind)
-            };
-        }
+            // Build a schema consisting of a single column.
+            var schemaBuilder = new SchemaBuilder();
+            schemaBuilder.AddColumn(scoreColumnName, scoreType, metadataBuilder.GetMetadata());
 
-        /// <summary>
-        /// Assumes all columns have ScoreColumnKind and ScoreValueKind of type Text.
-        /// </summary>
-        public virtual ColumnType GetMetadataTypeOrNull(string kind, int col)
-        {
-            Contracts.CheckNonEmpty(kind, nameof(kind));
-            Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-            switch (kind)
-            {
-            case MetadataUtils.Kinds.ScoreColumnKind:
-            case MetadataUtils.Kinds.ScoreValueKind:
-                return TextType.Instance;
-            }
-            return null;
+            return schemaBuilder.GetSchema();
         }
 
         /// <summary>
-        /// Assumes all columns have ScoreColumnKind and ScoreValueKind.
+        /// Create a <see cref="Schema"/> with two columns for binary classifier. The first column, indexed by 0, is the score column.
+        /// The second column is the probability column. For example, for linear support vector machine, score column stands for the inner product
+        /// of linear coefficients and the input feature vector and we convert score column to probability column using a calibrator.
         /// </summary>
-        public virtual void GetMetadata<TValue>(string kind, int col, ref TValue value)
+        /// <param name="scoreColumnName">Column name of score column</param>
+        /// <param name="probabilityColumnName">Column name of probability column</param>
+        /// <returns><see cref="Schema"/> of binary classifier's output.</returns>
+        public static Schema CreateBinaryClassificationSchema(string scoreColumnName = MetadataUtils.Const.ScoreValueKind.Score,
+            string probabilityColumnName = MetadataUtils.Const.ScoreValueKind.Probability)
         {
-            Contracts.CheckNonEmpty(kind, nameof(kind));
-            Contracts.CheckParam(0 <= col && col < ColumnCount, nameof(col));
-            switch (kind)
-            {
-            case MetadataUtils.Kinds.ScoreColumnKind:
-                ScoreColumnKindGetter.Marshal(col, ref value);
-                break;
-            case MetadataUtils.Kinds.ScoreValueKind:
-                ScoreValueKindGetter.Marshal(col, ref value);
-                break;
-            default:
-                throw MetadataUtils.ExceptGetMetadata();
-            }
-        }
+            // Schema of Score column. We are going to extend it by adding a Probability column.
+            var partialSchema = Create(NumberType.Float, MetadataUtils.Const.ScoreColumnKind.BinaryClassification, scoreColumnName);
```

Comment:
(minor) It feels a bit heavy-weight to create a whole schema object just to get 1 column out of it.

Reply:
This saves lines of creating metadata for that column (~6 lines). :)